libs/oauth-key-service/src/test/java/uk/gov/di/ipv/core/library/oauthkeyservice/OAuthKeyServiceTest.java

```diff
@@ -0,0 +1,210 @@
+        // Set up
+        when(mockHttpClient.send(any(HttpRequest.class), any(HttpResponse.BodyHandler.class)))
+                .thenReturn(mockHttpResponse);
+        when(mockHttpResponse.statusCode()).thenReturn(200);
+        when(mockHttpResponse.body())
+                .thenReturn(String.format("{\"keys\":[%s]}", RSA_ENCRYPTION_PUBLIC_JWK_NO_KEY_USE));
+
+        // Act
+        try (MockedStatic<HttpRequest.BodyPublishers> mockedBodyPublishers =
+                mockStatic(HttpRequest.BodyPublishers.class, CALLS_REAL_METHODS)) {
+            var key = oAuthKeyService.getEncryptionKey(oauthCriConfig);
+
+            // Assert
+            verify(mockHttpClient).send(httpRequestCaptor.capture(), any());
+            assertEquals(RSAKey.parse(TEST_KEY), key);
+        }
+    }
+
+    @Test
+    void getEncryptionKeyShouldReturnKeyFromConfigIfResponseReturnsEmptyArray() throws Exception {
+        // Set up
+        when(mockHttpClient.send(any(HttpRequest.class), any(HttpResponse.BodyHandler.class)))
+                .thenReturn(mockHttpResponse);
+        when(mockHttpResponse.statusCode()).thenReturn(200);
+        when(mockHttpResponse.body()).thenReturn("{\"keys\":[]}");
+
+        // Act
+        try (MockedStatic<HttpRequest.BodyPublishers> mockedBodyPublishers =
+                mockStatic(HttpRequest.BodyPublishers.class, CALLS_REAL_METHODS)) {
+            var key = oAuthKeyService.getEncryptionKey(oauthCriConfig);
+
+            // Assert
+            verify(mockHttpClient).send(httpRequestCaptor.capture(), any());
+            assertEquals(RSAKey.parse(TEST_KEY), key);
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {400, 500})
+    void getEncryptionKeyShouldReturnKeyFromConfigIfRequestErrors(int errorCode) throws Exception {
+        // Set up
+        when(mockHttpClient.send(any(HttpRequest.class), any(HttpResponse.BodyHandler.class)))
+                .thenReturn(mockHttpResponse);
+        when(mockHttpResponse.statusCode()).thenReturn(errorCode);
+
+        // Act
+        try (MockedStatic<HttpRequest.BodyPublishers> mockedBodyPublishers =
+                mockStatic(HttpRequest.BodyPublishers.class, CALLS_REAL_METHODS)) {
+            var key = oAuthKeyService.getEncryptionKey(oauthCriConfig);
+
+            // Assert
+            verify(mockHttpClient).send(httpRequestCaptor.capture(), any());
+            assertEquals(RSAKey.parse(TEST_KEY), key);
+        }
+    }
+
+    @Test
+    void getEncryptionKeyShouldReturnKeyFromConfigIfNoJwksUrl() throws Exception {
+        var oauthConfigNoJwksUrl = OauthCriConfig.builder().encryptionKey(TEST_KEY).build();
+        var key = oAuthKeyService.getEncryptionKey(oauthConfigNoJwksUrl);
+
+        assertEquals(RSAKey.parse(TEST_KEY), key);
+    }
+
+    @Test
+    void getEncryptionKeyShouldReturnCachedKeyIfItExistsAndIsNotExpired() throws Exception {
+        // Set up
+        // Create a valid key already in the cache
```

Comment:
Instead of doing this, I wonder if you could just call it twice and assert that the mockHttpClient is only called once? That's the public behaviour that we're trying to test - and the implementation doesn't really matter.

For the test below, you could do the same but set the config cache duration to zero...